Yoast/wordpress-seo
# Organize JavaScript-externals manage page dependencies

## Context


* Reduce the size of the JavaScript that is being loaded on page where it is not needed.
* Goals: quicker page-loads & less data transferred per page

### Todo

**Test with the addons**
* [ ] Premium - [PR](https://github.com/Yoast/wordpress-seo-premium/pull/2978)
* [ ] Local - only uses `yoastseo: window.yoast.analysis` as external - no adjustments needed
* [ ] WooCommerce - only uses `yoastseo: window.yoast.analysis` as external - no adjustments needed
* Video - doesn't use webpack
* News - doesn't use webpack

## Summary


This PR can be summarized in the following changelog entry:

* Improved the loading times of admin-pages where Yoast SEO places content.

## Relevant technical choices:

* Introduced 10 additional JavaScript-files to be loaded; though this helps to reduce the number of files that need to be loaded on *most* pages. The post-edit page already has ~60 files that are loaded (WordPress loads a lot of small files); adding 10 there did not seem like a big impact.
* Specified -all- dependencies for all assets declared in our admin-asset-manager (a lot were missing)
* Created a `components` which loads the dependencies that are currently tied to the `components-legacy` (yoast-components) asset
* Created a `components-modern` which is the `@yoast/components` package
* _More in the JavaScript repository-branch_

## Test instructions

This PR can be tested by following these steps:

* Visit all the pages we place javascript on and don't see any JavaScript errors and see that everything is loading/working as expected.
* WordPress Dashboard
* Post/term overview
* Post/term edit
* Settings pages
* Configuration wizard

## UI changes

## Documentation

## Quality assurance


Fixes PROD-202
Requires https://github.com/Yoast/javascript/pull/779
